modules/basicfuncs/tests/test_basicfuncs.c

```diff
@@ -153,6 +153,14 @@ test_str_funcs(void)
   assert_template_format("$(padding foo 10)", "       foo");
   assert_template_format("$(padding foo 10 x)", "xxxxxxxfoo");
   assert_template_format("$(padding foo 10 abc)", "abcabcafoo");
+
+  assert_template_failure("$(binary)", "Incorrect parameters");
+  assert_template_failure("$(binary abc)", "unable to parse abc");
+  assert_template_failure("$(binary 256)", "256 is above 255");
+  assert_template_format("$(binary 1)", "\1");
+  assert_template_format("$(binary 1 0x1)", "\1\1");
+  assert_template_format("$(binary 0xFF 255 0377)", "\xFF\xFF\xFF");
+  assert_template_format_with_len("$(binary 0xFF 0x00 0x30)", "\xFF\0000", 3);
```

Comment:
The expected value should be \xFF\0030?

Reply:
No, \000 is an octal zero, 0 is a character, which is the same as 0x30. But I have
now changed 0x30 to 0x40, which is a '@', should be easier to grasp.